modules/ui/commit_warnings.js

```diff
@@ -1,103 +1,113 @@
-
+            if (!issues.length) continue;
 
-            var items = container.select('ul').selectAll('li')
-                .data(issues, function(d) { return d.key; });
-
-            items.exit()
-                .remove();
-
-            var itemsEnter = items.enter()
-                .append('li')
-                .attr('class', issueItem);
-
-            var buttons = itemsEnter
-                .append('button')
-                .on('mouseover', function(d3_event, d) {
-                    if (d.entityIds) {
-                        context.surface().selectAll(
-                            utilEntityOrMemberSelector(
-                                d.entityIds,
-                                context.graph()
-                            )
-                        ).classed('hover', true);
-                    }
+            // Create a collapsible section for each severity level
+            var section = uiSection('issues-' + severity, context)
+                .label(() => {
+                    var count = issues.length;
+                    return t.append(
+                        'inspector.title_count',
+                        { title: t('issues.' + severity + 's.list_title'), count: count }
+                    );
                 })
-                .on('mouseout', function() {
-                    context.surface().selectAll('.hover')
-                        .classed('hover', false);
+                .disclosureContent(function(selection) {
+                    return renderIssuesList(selection, severity, issues);
                 })
-                .on('click', function(d3_event, d) {
-                    context.validator().focusIssue(d);
+                .shouldDisplay(function() {
+                    return issues && issues.length;
                 });
 
-            buttons
-                .call(svgIcon('#iD-icon-alert', 'pre-text'));
-
-            buttons
-                .append('strong')
-                .attr('class', 'issue-message');
-
-            buttons.filter(function(d) { return d.tooltip; })
-                .call(uiTooltip()
-                    .title(function(d) { return d.tooltip; })
-                    .placement('top')
-                );
-
-            items = itemsEnter
-                .merge(items);
-
-            items.selectAll('.issue-message')
-                .text('')
-                .each(function(d) {
-                    return d.message(context)(d3_select(this));
-                });
+            // Add the appropriate class for styling based on severity
+            selection
+                .call(section.render)
+                .classed(severity + '-section', true);
         }
     }
 
+    function renderIssuesList(selection, severity, issues) {
+        selection.selectAll('.issues-list').remove();
+        var container = selection
+            .append('ul')
+            .attr('class', 'changeset-list');
+
+        container.exit().remove();
```

Comment:
```suggestion
        let container = selection
            .selectAll('.changeset-list')
            .data([0]);

        container = container.enter()
            .append('ul')
            .attr('class', 'changeset-list')
            .merge(container);
```

Notice the `.data([0])` here: this assigns a constant `0` as a kind of _dummy data_ to the list: this results in the list only to only be created once (in `.enter().append(…)`) and only if the selection is empty.